src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -3,18 +3,28 @@
             [puppetlabs.services.protocols.request-handler :as handler]
             [puppetlabs.services.request-handler.request-handler-core :as core]
             [puppetlabs.services.jruby.jruby-puppet-service :as jruby]
-            [puppetlabs.trapperkeeper.services :as tk-services]))
+            [puppetlabs.trapperkeeper.services :as tk-services]
+            [clojure.tools.logging :as log]))
 
 (defn- handle-request
-  [request jruby-service config]
-  (jruby/with-jruby-puppet jruby-puppet jruby-service
-    (core/handle-request request jruby-puppet config)))
+  [request jruby-service config puppet-version]
+  (try
+    (jruby/with-jruby-puppet jruby-puppet jruby-service
+                             (core/handle-request request jruby-puppet config))
+    (catch IllegalStateException e
```

Comment:
Yeah, we'd definitely have to have the try/catch block wrapped around the invocation of `with-jruby-puppet` in order to be able to catch re-throw for a borrow timeout failure.  The only part I was debating was whether it would make sense to take the code in this function and push it down into the `request-handler-core` namespace or just leave it here - i.e., does this kind of logic more generally feel like something that should live in the *core* vs. the *service* namespace.  I'd be fine leaving it here - no strong opinion.  Maybe @cprice404 or @KevinCorcoran would have a preference?